server/routes/api/browser.py

```diff
@@ -110,3 +110,52 @@ def get_observation_ids():
         dcid = cells[0].get('value', '')
         result[obsDate] = dcid
     return Response(json.dumps(result), 200, mimetype='application/json')
+
+
+def statvar_hierarchy_helper(svg_id, sv_groups, processed_svg, processed_sv,
+                             seen_nodes):
+    sv_group = sv_groups.get(svg_id, {})
+    for child_sv in sv_group.get("childStatVars", []):
+        if child_sv in seen_nodes:
+            continue
+        child_sv_node = {}
+        child_sv_node["parent"] = svg_id
+        processed_sv[child_sv] = child_sv_node
+        seen_nodes.add(child_sv)
+    for child_svg in sv_group.get("childStatVarGroups", []):
+        child_svg_id = child_svg.get("id")
+        child_svg_node = sv_groups[child_svg_id]
+        if not "parent" in child_svg_node:
+            child_svg_node["parent"] = []
+        child_svg_node["parent"].append(svg_id)
+        processed_svg[child_svg_id] = child_svg_node
```

Comment:
processed_svg -> processed_svg_map; so this corresponding to the input "svg_map"

Reply:
done